File: solutions/python/word-count/1/word_count.py
def clean_text(text):
    punctuation = "!@#$%^&*()-_=+:.,"
    for p in punctuation:
        text = text.replace(p, " ")
    if text.startswith("'"):
        text = text[1:]
    if text.endswith("'"):
        text = text[:-1]
    return text.replace("\n", " ")

def count_words(sentence):
    result = {}
    for word in clean_text(sentence).split():
        cleaned_word = clean_text(word).lower()
        if cleaned_word in result:
            result[cleaned_word] += 1
        else:
            result[cleaned_word] = 1
    return result

File: solutions/csharp/word-count/1/WordCount.cs
public static class WordCount {
  public static IDictionary<string, int> CountWords(string phrase) {
    var result = new Dictionary<string, int>();
    foreach (var word in CleanPhrase(phrase)) {
      var cleanedWord = word.Trim('\'');
      if (string.IsNullOrWhiteSpace(cleanedWord)) {
        continue;
      }

      result[cleanedWord] = result.ContainsKey(cleanedWord)
        ? result[cleanedWord] + 1
        : 1;
    }

    return result;
  }

  private static IEnumerable<string> CleanPhrase(string phrase) {
    return string.Join("",
      phrase.Select(c => char.IsLetterOrDigit(c) || c == '\''
          ? char.ToLower(c)
          : ' '))
        .Split(" ");
  }
}